src/tiled/tileanimationeditor.cpp

```diff
@@ -392,6 +410,26 @@ void TileAnimationEditor::framesEdited()
     mApplyingChanges = false;
 }
 
+void TileAnimationEditor::setFrameTime()
+{
+    QItemSelectionModel *selectionModel = mUi->frameList->selectionModel();
+    const QModelIndexList indexes = selectionModel->selectedIndexes();
+
+    if (indexes.isEmpty()) {
+        mFrameListModel->setDefaultFrameTime(mUi->frameTime->value());
+        mUi->frameList->setFocus(); // This is necessary for the UI, It does not get updates without this
```

Comment:
This is a hack, which works because it triggers a repaint. The reason the UI isn't updating is that `setDefaultFrameTime` is modifying the frame durations without emitting the `dataChanged` signal.

Please write a separate function like `FrameListModel::applyFrameTime` that sets the duration like currently `setDefaultFrameTime` is doing, but in addition it should emit `dataChanged` with appropriate indexes. Something like:
```c++
emit dataChanged(index(0), index(mFrames.length() - 1), { Qt::EditRole, Qt::DisplayRole });
```
(since this will only be valid when there is at least one frame, you need to early-out if `mFrames.isEmpty()`)

Reply:
I now understood why it wasn't working and why the hack worked. I am going to use the function that was written previously. It internally emits a data change signal mFrameListModel->setData()